packages/rpc-core/src/rpc-methods/__tests__/get-blocks-with-limit-test.ts

```diff
@@ -0,0 +1,63 @@
+import { createHttpTransport, createJsonRpc } from '@solana/rpc-transport';
+import type { Rpc } from '@solana/rpc-transport/dist/types/json-rpc-types';
+import fetchMock from 'jest-fetch-mock-fork';
+
+import { createSolanaRpcApi, SolanaRpcMethods } from '../index';
+
+describe('getBlocksWithLimit', () => {
+    let rpc: Rpc<SolanaRpcMethods>;
+    beforeEach(() => {
+        fetchMock.resetMocks();
+        fetchMock.dontMock();
+        rpc = createJsonRpc<SolanaRpcMethods>({
+            api: createSolanaRpcApi(),
+            transport: createHttpTransport({ url: 'http://127.0.0.1:8899' }),
+        });
+    });
+
+    describe('when called with a valid `startSlot` and a valid `limit`', () => {
+        it('returns 8 blocks for a limit of 8', async () => {
+            expect.assertions(1);
+            const res = await rpc.getBlocksWithLimit(5n, 8).send();
+            expect(res).toHaveLength(8);
```

Comment:
Does this only work if there are >13 blocks at the point this test runs? Just wondering if it's potentially going to be flaky in CI if this test ends up running very soon after the test validator starts

Reply:
Ahh yeah good point. Could just add a sleep timer then?